controllers/idler/idler_controller.go

```diff
@@ -132,7 +132,13 @@ func (r *Reconciler) ensureIdling(logger logr.Logger, idler *toolchainv1alpha1.I
 			// Already tracking this pod. Check the timeout.
 			if time.Now().After(trackedPod.StartTime.Add(time.Duration(idler.Spec.TimeoutSeconds) * time.Second)) {
 				podLogger.Info("Pod running for too long. Killing the pod.", "start_time", trackedPod.StartTime.Format("2006-01-02T15:04:05Z"), "timeout_seconds", idler.Spec.TimeoutSeconds)
-
+				var podreason string
+				podcondition := pod.Status.Conditions
+				for _, podtype := range podcondition {
+					if podtype.Type == "Ready" {
+						podreason = podtype.Reason
+					}
+				}
```

Comment:
```suggestion
		readyCondition, found := condition.FindConditionByType(pod.Status.Conditions, toolchainv1alpha1.ConditionReady)
               if found {
                    podReason := readyCondition.Reason
                }
```

could maybe do something like this using the common function - https://github.com/codeready-toolchain/toolchain-common/blob/master/pkg/condition/condition.go#L51

Reply:
Here i am not able to use this function as the variable pod.Status.condition,  the info on pod status we get it from corev1 and not toolchainv1alpha1

Reply:
you're right, its the podCondition here, not condition :)